website/src/models/challenge.js

```diff
@@ -97,7 +97,6 @@ schema.methods.syncToUser = async function syncChallengeToUser (user) {
   let [challengeTasks, userTasks] = await Q.all([
     // Find original challenge tasks
     Tasks.Task.find({
-      userId: {$exists: false},
```

Comment:
The userId property can be found on all tasks and it's used to link a task to a user and in case it's missing to say that this task doesn't belong to a specific user (but to a challenge).

What we're doing here are 2 queries:
- one to fetch all the tasks belonging to a challenge and not to any user (the original challenge tasks, `chalTasks`)
- the second to fetch all tasks belonging to our user AND to the challenge.

This way we can see if any new task has been added to the challenge but not yet synced to the user (and add it to the user), deleted from the challenge but not yet synced to the user (and remove it from the user) and if any challenge task has been edited and needs to be synced to the user.

If you remove the `userId: {$exists: false}` query modified all challenge tasks will be fetched, the ones belonging to the challenge (which we use to compare to the user's tasks) but also all the challenge tasks from all the users that are members of the challenged which we don't need.

Let me know if I'm not being very clear